### PR TITLE
Internal first version

### PR DESCRIPTION
**Pull request by kazuhitoyokoi**
I created the internal first version of the Node-RED desktop. This version has a window and installer for Windows and macOS.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      "publish": []
+    }
+  },
+  "scripts": {
+    "start": "electron .",
+    "build": "electron-builder"
+  },
+  "dependencies": {
+    "node-red": "1.1.3",
+    "node-red-dashboard": "^2.23.3"
```
Comment (knolleary) [inline package.json]: Should we be including dashboard by default?

File: README.md [hunk 2]
```diff
@@ -1 +1,27 @@
 # node-red-desktop
+The Node-RED desktop is a desktop application which includes Node-RED, Node.js, nodes, and Chromium browser engine.
```
Comment (knolleary) [inline README.md]: This needs to make it clear this project is under development and **not** ready for general use. The Quick Start instructions are not valid until we do a release.

File: main.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+if (!app.requestSingleInstanceLock()) {
+    app.quit();
+} else {
+    RED.init(server, settings);
+    expressApp.use(settings.httpAdminRoot, RED.httpAdmin);
+    expressApp.use(settings.httpNodeRoot, RED.httpNode);
+    server.listen(settings.uiPort);
+    RED.start().then(function () {
+        app.whenReady().then(createWindow);
+    });
```
Comment (knolleary) [inline main.js:102]: This will need error handling - such as `port in use` and all the other reasons that `RED.start()` might fail.

**Review by knolleary — CHANGES_REQUESTED** [review]
I understand this is just the first commit to get something in, but there are a couple changes to make.

Please also ensure the standard `.github` templates are added to the repository - you can copy them from the main repository - https://github.com/node-red/node-red/tree/master/.github (ignore the `scripts` and `workflows` directories).

There should also be `SECURITY.md` and `CONTRIBUTING.md` copied over from the main repo.

File: package.json (hunk 1 above)
Comment (dceejay) [reply package.json]: Good point - no - not if this is focussed around the editor.

Comment (kazuhitoyokoi) [reply package.json]: I agree with that. I removed the dashboard from the dependency modules.

File: SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Security Policy
+
+## Supported Versions
+
+| Version | Supported          |
+| ------- | ------------------ |
+| 1.0.0   | :white_check_mark: |
+| 0.20.x  | :white_check_mark: |
```
Comment (dceejay) [inline SECURITY.md]: What do these refer to ? presumably versions of Node-RED - or is it this project ?

Comment (knolleary) [reply SECURITY.md]: I asked @kazuhitoyokoi to copy over the standard SECURITY.md file from the main repository. I forgot that had this table. I was more concerned with having the information about how to report security issue - this table can be removed from the non-core copies.

Comment (kazuhitoyokoi) [reply SECURITY.md]: I learned what is the SECURITY.md file. I removed the table from the file.

File: main.js (hunk 3 above)
Comment (kazuhitoyokoi) [reply main.js:102]: @knolleary Thank you for your advice. I added the error handling when starting Node-RED. Users can see the error from the dialog window.

**Pull request merged**